packages/react/src/domain/types.ts

```diff
@@ -18,6 +18,10 @@ export function isFactorOidc(factor: Factor): factor is FactorOIDC {
   return factor.method === "oidc";
 }
 
+export function hasOidcAndNonOidcFactors(factors: Factor[]): boolean {
```

Comment:
This is a bit pedantic, not a blocker - a good way of documenting this function would be writing a couple of unit tests that show what the expected behavior is. I would write a test with three cases, the same way you did for the component:
- returns true if both oidc and non oidc are present
- returns false if no oidc factors
- returns false if no non-oidc factors

PR is approved anyway, up to you if you want to add this!